docs/samples/gcp-iap/README.md

```diff
@@ -0,0 +1,137 @@
+## KFServing GCP/IAP Example 
+When using Kubeflow with GCP it is common to use a [GCP Identity Aware Proxy](https://cloud.google.com/iap) (IAP) to manage client authentication to the KFServing endpoints.  The proxy intercepts and authenticates users and passes identity assertion (JWT) to kubernetes service/pods.  Whilst it is also possible to add access control (i.e. programmable or service mesh authorization), this is not described here.
+
+### Prerequisites
+1. Your ~/.kube/config should point to a cluster with [KFServing installed](https://github.com/kubeflow/kfserving/blob/master/docs/DEVELOPER_GUIDE.md#deploy-kfserving).
+2. Your gcloud config is initialised to the project containing the k8s cluster and has a service-account that can download IAP key file.
+2. Your k8s cluster has a GCP external http loadbalancer (i.e. you've configured an ingress gateway).
+3. You are using a fairly recent version of istio (istio-networking is v0.11.2+ )
+4. You are using a recent version of KFServing (v0.3+)
+
+### Overview
+This example shows how the [sklearn iris sample](https://github.com/kubeflow/kfserving/tree/master/docs/samples/sklearn) can be adapted to support GCP IAP.  We will use a pre-trained model that is hosted on a public gcs bucket.  The model predicts a binary classificaton of iris specie using 4 numerical features.
+
+In this example we will:
+ 1. **Enable GCP IAP**
+ 1. **Change firewall rules (for private k8s clusters)**
+ 1. **Create the inference service**
+ 1. **Test the private predict endpoint (using port-forwarding)**
+ 1. **Expose the inference service externally using an additional Istio Virtual Service**
+ 1. **Test the external predict endpoint (using iap_request.py)**
+
+Step 5. will become unnecessary when this [kfserving issue](https://github.com/kubeflow/kfserving/issues/824) is resolved.
+
+### Enable GCP IAP for k8s
+You have followed the steps to [Set up OAuth for Cloud IAP](https://www.kubeflow.org/docs/gke/deploy/oauth-setup/)
+
+We assume that you have created Kubeflow profiles and that [GCP workload identity](https://www.kubeflow.org/docs/gke/authentication/) is configured.
+
+This may not be essential for this example, but is recommended as it maps GCP service-accounts (understood by the IAP) to kubernetes service-accounts (understood by kubernetes and istio RBAC).
+
+### Change firewall rules (for private k8s clusters) 
+If you are running Kubeflow in a [private GKE cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters) it will be configured with a restrictive firewall.  
+You will need to ensure that the k8s master node(s) can talk to the following k8s worker node(s) ports: 
+ - 6443 (for cert-manager) 
+ - 8443 (for probes such as kfserving)
```

Comment:
what's this probe ?

Reply:
OK Kfserving probe is the wrong term but the 8433 port is required.
Without it I get a timeout after ~30s
```rror from server (InternalError): error when creating "sklearn-iap-no-authz.yaml": Internal error occurred: failed calling webhook "inferenceservice.kfserving-webhook-server.defaulter": Post https://kfserving-webhook-server-service.kubeflow.svc:443/mutate-inferenceservices?timeout=30s: dial tcp 10.36.1.33:9876: i/o timeout```
I know it mentions 443 but allowing access to 8433 resolves the issue.
I'm going to change it to 'istio readiness probe'.  When I was talking to you about it on slack you mentioned PR #1137.  
I think opening 8443 with private gke cluster is useful for a bunch of stuff tbh e.g. prometheus
https://medium.com/@damianmyerscough/private-gke-cluster-running-cert-manager-prometheus-operator-57b676406d9a

Reply:
I'm going to mark this as resolved - feel free to suggest a better name than 'istio readiness probe'